### PR TITLE
Add Account shared components

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?
Add Account shared components

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/components/Account/components/ContentWrapper.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          )}
+          {children(this.handleError)}
+        </main>
+      </section>
+    )
+  }
+}
+
+ContentWrapper.propTypes = {
+  intl: intlShape,
```
Comment (kevinch) [inline react/components/Account/components/ContentWrapper.js]: Keep consistency in project, have this prop always _required_ or not.

Comment (salesfelipe) [reply react/components/Account/components/ContentWrapper.js]: 👍

File: react/components/Account/components/BaseLoading.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    const { isLoading } = this.state
+    const { headerConfig, children, queryData } = this.props
+
+    const hasAuthenticationError =
+      queryData.error &&
+      queryData.error.toString().indexOf('not authenticated') > -1
+
+    return (
+      <ContentWrapper {...headerConfig}>
+        {() => (
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply react/components/Account/components/BaseLoading.js]: Unfortunately yes 😢

File: react/Account.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+export { default as default } from './components/Account'
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply react/Account.js]: yes

File: react/components/Account/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  {children}
+</BaseLoading>
+```
+
+```jsx
+<GenericError onDismiss={handleError} errorId={error} />
+```
+
+TODO
+
```
Comment (kevinch) [inline react/components/Account/README.md]: Empty section?

**Comment by firstdoit** [comment]
⚠️  You released and didn't merge!

File: react/components/Account/components/BaseLoading.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          <Fragment>
```
Comment (lucasecdb) [inline react/components/Account/components/BaseLoading.js]: Why do you need this?

File: react/components/Account/components/ContentWrapper.js [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+  intl: intlShape.isRequired,
+  children: PropTypes.func.isRequired,
+  title: PropTypes.string,
+  titleId: PropTypes.string,
+  backButton: PropTypes.shape({
+    title: PropTypes.string,
+    titleId: PropTypes.string,
+    path: PropTypes.string.isRequired,
+  }),
+  history: ReactRouterPropTypes.history.isRequired,
```
Comment (lucasecdb) [inline react/components/Account/components/ContentWrapper.js]: `ReactRouter`?

File: react/package.json [hunk 7]
```diff
@@ -23,6 +23,8 @@
     "react-motion": "^0.5.2",
     "react-resize-detector": "^3.0.1",
     "react-responsive-modal": "^2.0.1",
+    "react-router-dom": "^4.3.1",
```
Comment (lucasecdb) [inline react/package.json]: 👀

File: react/components/Account/components/BaseLoading.js (hunk 5 above)
Comment (salesfelipe) [reply react/components/Account/components/BaseLoading.js]: https://github.com/vtex-apps/my-account/blob/085da0677e9bec38ce12b5127fda5823be57850d/react/components/loaders/AddressCreateLoading.js#L9

File: react/components/Account/components/ContentWrapper.js (hunk 6 above)
Comment (salesfelipe) [reply react/components/Account/components/ContentWrapper.js]: The my-account app is basically another SPA for itself, the react-router is needed so make the communication between the pages inside the app.

File: react/components/Account/README.md [hunk 8]
```diff
@@ -16,6 +18,17 @@ You can use it in your code like a React component with the jsx tag: `<ContentWr
 </ContentWrapper>
 ```
 
+
+| Prop name          | Type       | Description                                                                 |
```
Comment (lucasecdb) [inline react/components/Account/README.md]: I think it's better to not write out these spaces, it makes it harder to change later.

**Pull request merged**